crates/cranelift/src/gc/enabled/drc.rs

```diff
@@ -105,6 +117,47 @@ impl DrcCompiler {
         (activations_table, next, end)
     }
 
+    /// Write to an uninitialized field or element inside a GC object.
+    fn init_field(
+        &mut self,
+        func_env: &mut FuncEnvironment<'_>,
+        builder: &mut FunctionBuilder<'_>,
+        field_addr: ir::Value,
+        ty: &WasmStorageType,
+        val: ir::Value,
+    ) -> WasmResult<()> {
+        // Data inside GC objects is always little endian.
+        let flags = ir::MemFlags::trusted().with_endianness(ir::Endianness::Little);
+
+        match ty {
+            WasmStorageType::Val(WasmValType::Ref(r))
+                if r.heap_type.top() == WasmHeapTopType::Func =>
+            {
+                return Err(wasm_unsupported!(
+                    "funcrefs inside the GC heap are not yet implemented"
+                ));
+            }
```

Comment:
Could `translate_init_gc_reference` below internally do this so it doesn't have to be done each time a translate helper is called? (I saw this above as well for writes I think)

Reply:
I actually like having it repeated, so I know exactly all the places I need to fix up later (which I plan on doing soon, so that I can run some of the spec tests that use funcrefs)